content/en/blog/improving-observability-opentelemetry-datadog-fission/index.md

```diff
@@ -0,0 +1,288 @@
+
+```yaml
+export FISSION_NAMESPACE=fission
+helm install --namespace $FISSION_NAMESPACE \
+fission fission-charts/fission-all \
+--set openTelemetry.otlpCollectorEndpoint="otel-collector.opentelemetry-operator-system.svc:4317" \
+--set openTelemetry.otlpInsecure=true \
+--set openTelemetry.tracesSampler="parentbased_traceidratio" \
+--set openTelemetry.tracesSamplingRate="1"
+
+```
+
+### OpenTelemetry Configuration
+
+To send our traces to Datadog from OpenTelemetry, we first need to make use of `OpenTelemetry Collector` that will help collect the traces.
+It will internally share the traces with an `OpenTelemetry Processor` that will eventually share it with an `OpenTelemetry Exporter` that will finally send the traces to Datadog.
+
+To manage all of this, OpenTelemetry provides us with an `OpenTelemetry Operator`.
+The operator can be used to provision and maintain OpenTelemetry Collector instance along with handling its configuration changes.
+However, to use the operator in an existing cluster, we need to add a `cert-manager`.
+
+You can use the following commands to add `cert-manager` to your existing cluster and install the OpenTelemetry Operator:
+
+```bash
+# cert-manager
+kubectl apply -f https://github.com/jetstack/cert-manager/releases/latest/download/cert-manager.yaml
+
+# open telemetry operator
+kubectl apply -f https://github.com/open-telemetry/opentelemetry-operator/releases/latest/download/opentelemetry-operator.yaml
+```
+
+
+The `OpenTelemetry Collector` is a vendor agnostic agent that can send and receive metrics, traces to different tools in different formats.
+In order to use it with Fission functions, we need to modify the `OpenTelemetry Collector` configuration as follows.
+
+Under exporters add `Datadog` and make sure to update your `API KEY` in the following yaml configuration.
+
+```yaml
+...
+...
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: otel-collector-conf
+  labels:
+    app: opentelemetry
+    component: otel-collector-conf
+data:
+  otel-collector-config: |
+    receivers:
+      otlp:
+        protocols:
+          grpc:
+          http:
+    processors:
+      batch:
+        timeout: 10s
+      k8sattributes:
+    extensions:
+      health_check: {}
+      zpages: {}
+    exporters:
+      datadog:
+        api:
+          key: <YOUR_API_KEY>
+    service:
+      telemetry:
+        logs:
+          level: "DEBUG"
+      extensions: [health_check, zpages]
+      pipelines:
+        metrics/2:
+          receivers: [otlp]
+          processors: [batch, k8sattributes]
+          exporters: [datadog]
+        traces/2:
+          receivers: [otlp]
+          processors: [batch, k8sattributes]
+          exporters: [datadog]```
+...
+...
+```
+
+> Note: The complete yaml configuration that you need to apply for `OpenTelemetry` to work correctly can be found in the `optel-config.yaml` file [here](https://github.com/fission/examples/blob/master/python/opentelemetry-datadog/optel-config.yaml).
```

Comment:
We can do link assignment to file name instead of `here`

Reply:
done